### PR TITLE
CATROID-577 "Get image from ___ and use as current look" brick

### PR DESCRIPTION
**Pull request by mandelbrot2357**
Added new bricks to retrieve image from URL and set as look or background.

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/ui/fragment/CategoryBricksFactory.java [hunk 1]
```diff
@@ -478,6 +480,16 @@
...
 		}
 
+		if (BuildConfig.FEATURE_WEBREQUEST_BRICK_ENABLED) {
+			if (!isBackgroundSprite) {
+				looksBrickList.add(new LookRequestBrick(context.getString(R.string.brick_look_request_default_value)));
+			} else if (ProjectManager.getInstance().getCurrentProject().getXmlHeader().islandscapeMode()) {
+				looksBrickList.add(new BackgroundRequestBrick(context.getString(R.string.brick_background_request_default_value_landscape)));
+			} else {
+				looksBrickList.add(new BackgroundRequestBrick(context.getString(R.string.brick_background_request_default_value_portrait)));
+			}
```
Comment (DinosaurierRex) [inline catroid/src/main/java/org/catrobat/catroid/ui/fragment/CategoryBricksFactory.java:490]: The `BackgroundRequestBrick` will only be shown for Background Sprites. So its not possible to set the background from another Script. Is this specified like this? Because I tried and it would work to set the background from other scripts.

File: catroid/src/main/java/org/catrobat/catroid/utils/Utils.java [hunk 2]
```diff
@@ -195,6 +198,27 @@ public static String generateRemixUrlsStringForMergedProgram(XmlHeader headerOfF
 		return extractedUrls;
 	}
 
+	public static String getFileNameFromURL(String url) {
+		int extension = url.lastIndexOf('.');
+		int path = Math.max(url.substring(0, extension).lastIndexOf('/'), 0) + 1;
+		int param1 = url.indexOf('&', extension);
+		int param2 = url.indexOf('?', extension);
+		int param = (param2 >= 0 && param2 < param1) || (param1 == -1) ? param2 : param1;
+		return param >= 0 ? url.substring(path, param) : url.substring(path);
```
Comment (DinosaurierRex) [inline catroid/src/main/java/org/catrobat/catroid/utils/Utils.java:207]: Could you please think of better namings here. This is not clean code like and hard to follow what its doing.

File: catroid/src/main/java/org/catrobat/catroid/ui/fragment/CategoryBricksFactory.java (hunk 1 above)
Comment (mandelbrot2357) [reply catroid/src/main/java/org/catrobat/catroid/ui/fragment/CategoryBricksFactory.java:490]: Yes, that's by design. The brick is only supposed to be visible for background sprites.

**Comment by wslany** [comment]
Thanks!!
